src/mongocxx/test/uri.cpp

```diff
@@ -295,6 +313,228 @@ TEST_CASE("uri::wait_queue_timeout_ms()", "[uri]") {
+                                   std::mutex& mtx,
+                                   std::vector<std::string>& new_hosts) {
+    using namespace mongocxx;
+
+    auto admin = client["admin"];
+    auto result = admin.run_command(make_document(kvp("ping", 1)));
+    REQUIRE(result.begin()->get_double() == 1.0);
+
+    auto uri = client.uri();
+    auto my_options = uri.options();
+    auto creds = uri.credentials();
+    auto expected_options = test_doc["options"];
+    if (expected_options) {
+        compare_options(expected_options.get_document().view(), my_options, creds);
+    }
+
+    auto expected_hosts = test_doc["hosts"];
+    if (expected_hosts) {
+        size_t count = 0;
+        while (!hosts_are_equal(expected_hosts.get_array().value, new_hosts, mtx)) {
+            count++;
+            if (count > 4) {
+                REQUIRE(hosts_are_equal(expected_hosts.get_array().value, new_hosts, mtx));
+            }
+            std::this_thread::sleep_for(std::chrono::seconds(count));
+        }
+    }
+}
+
+static void run_srv_max_hosts_test_file(bsoncxx::document::view test_doc) {
+    using namespace mongocxx;
+
+    bool expect_exception = test_doc["error"] && test_doc["error"].get_bool().value;
+    bool should_ping = true;
+    if (test_doc["ping"] && test_doc["ping"].get_bool() == false) {
+        should_ping = false;
+    }
+    should_ping = should_ping && !expect_exception;
+
+    options::apm apm_opts;
+    std::mutex mtx;
+    std::vector<std::string> new_hosts;
+
+    apm_opts.on_topology_changed([&](const events::topology_changed_event& event) {
+        const std::lock_guard<std::mutex> lock(mtx);
+        auto new_td = event.new_description();
+        auto servers = new_td.servers();
+        new_hosts.clear();
+        for (const auto& i : servers) {
+            new_hosts.push_back(std::string(i.host()) + ":" + std::to_string(i.port()));
+        }
+    });
+
+    mongocxx::options::tls tls_options;
+    mongocxx::options::client client_options;
+
+    auto ca_file_path = std::getenv("MONGOCXX_TEST_TLS_CA_FILE");
+    REQUIRE(ca_file_path);
+    tls_options.ca_file(ca_file_path);
+    tls_options.allow_invalid_certificates(true);
+    client_options.tls_opts(tls_options);
+    client_options.apm_opts(apm_opts);
+
+    mongocxx::uri my_uri;
+    try {
+        my_uri = uri{test_doc["uri"].get_string().value};
+        class client client {
+            my_uri, test_util::add_test_server_api(client_options)
+        };
+        bool using_tls = client.uri().tls();
+        if (!using_tls) {
+            return;
+        }
+        REQUIRE(!expect_exception);
+        if (should_ping) {
+            validate_srv_max_hosts(std::move(client), test_doc, mtx, new_hosts);
+        }
+    } catch (mongocxx::exception& e) {
+        REQUIRE(expect_exception);
+    }
+}
+
+static void iterate_srv_max_hosts_tests(std::string dir, std::vector<std::string> files) {
```

Comment:
`src/mongocxx/test/uri.cpp` is reserved for URI tests that do not require a live server and is part of `test_driver`. For consistency with other unified spec tests, suggest moving these tests (that are not testing the URI format itself, but testing behavior against a live server) into a new `src/mongocxx/test/spec/uri.cpp` file (see other test files in that directory for reference) as part of a new `test_uri_options` test executable (see [other executables](https://github.com/mongodb/mongo-cxx-driver/blob/cb3193179199bd764ff9706f3a5fc19f5d8f4349/src/mongocxx/test/CMakeLists.txt#L106) for reference).